xlive/Blam/Engine/game/players.cpp

```diff
@@ -157,3 +157,132 @@ uint32 player_appearance_required_bits()
 {
 	return 39;
 }
+
+void __cdecl player_validate_configuration(datum player_index, s_player_properties* configuration_data)
+{
+    // Campaign verification
+    if (game_is_campaign())
+    {
+        scenario* scnr = get_global_scenario();
+        uint32 block_size = scnr->player_starting_locations.size;
+        if (block_size > 0)
+        {
+            uint32 i = 0;
+            while (true)
```

Comment:
Are you sure you're not missing any breaks here, in this loop?

Reply:
Also why not a normal `for` loop?

Reply:
Changed to a for loop.